language/move-vm/types/src/values/values_impl.rs

```diff
@@ -474,8 +474,139 @@ impl ValueImpl {
 
             (ContainerRef(l), ContainerRef(r)) => l.equals(r)?,
             (IndexedRef(l), IndexedRef(r)) => l.equals(r)?,
-
-            _ => {
+            (Invalid, Invalid)
```

Comment:
I'd probably just do an exhaustive match for one side, `(Invalid, _) | (U8(_), _) | (U16(_), _) | ... `